# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/generic/HyperTextAccessible.cpp
# Commit: 59693d9d9f60
# Full Hash: 59693d9d9f6036e10fe4656e9c86f52749e8e827
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2019-09-30 21:54:36
# Description:
#   Bug 1584465: Null check parent accessible before attribute check to cover root case. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47488
# ==============================================================================

diff -r ab8923caa594 -r 59693d9d9f60 accessible/generic/HyperTextAccessible.cpp
--- a/accessible/generic/HyperTextAccessible.cpp	Mon Sep 30 14:59:01 2019 +0000
+++ b/accessible/generic/HyperTextAccessible.cpp	Mon Sep 30 16:23:30 2019 +0000
@@ -300,7 +300,7 @@
       // bulleted lists here because PeekOffset returns the inner text node
       // for a list when it should return the list bullet.
       // We manually set the offset so the error doesn't propagate up.
-      if (offset == 0 && descendant->Parent()->IsHTMLListItem() &&
+      if (offset == 0 && parent && parent->IsHTMLListItem() &&
           descendant->PrevSibling() && descendant->PrevSibling()->GetFrame() &&
           descendant->PrevSibling()->GetFrame()->IsBulletFrame()) {
         offset = 0;
